flink-staging/flink-ml/src/main/scala/org/apache/flink/ml/tree/Tree.scala

```diff
@@ -0,0 +1,327 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.tree
+
+import java.util
+
+import org.apache.flink.ml.math.{ContinuousHistogram, OnlineHistogram, Vector}
+
+import scala.collection.mutable
+
+/** Tree structure. This is kind of maintained in an unconventional way.
+  * We provide direct access to all nodes
+  * The obvious assumption is that child of i are 2*i and 2*i+1, while parent of i is i/2
+  *
+  */
+class Tree(
```

Comment:
Formatting. We try to follow the rules as presented [here](https://github.com/databricks/scala-style-guide#indent)